### DISCUSSION TITLE
read a file which is pex-ed

### DISCUSSION THREAD

**Discussion by ws1088** [pex-tool/pex#1373] [Q&A]
I have verified that my data file is zipped in the .pex archive. But how do I refer to it in my Python code? I also need to make sure that the code works with non-pex run as well (which means I need a way to determine if i am running from within pex-ed executable. pyinstaller does have this feature).

**Comment by ws1088** [comment]
found out that i can use `--not-zip-safe` option. is there any other way? thanks!

**Reply by jsirois** [reply]
Sorry for the late reply! Since #1438 (2 months after you asked this question) all PEXed user and third party code runs fully extracted to disk, the `--not-zip-safe` is now a no-op:
```
$ pex -h | grep -A3 zip-safe
  --zip-safe, --not-zip-safe
                        Deprecated: This option is no longer used since user
                        code is now always unzipped before execution.
                        (default: None)
```
